### ISSUE TITLE
GM_xmlhttpRequest not working anymore on Chrome in Incognito

### ISSUE THREAD

**Issue by jdoe1024** [Tampermonkey/tampermonkey#1220]
Using `GM_xmlhttpRequest()` used to work correctly with previous version of Chrome, but now, there is an error in the extension background page console for most (if not all) requests in Incognito mode:

> Access to XMLHttpRequest at 'https://example.net/' from origin 'chrome-extension://dhdgffkkebhmkfjojejmpbldmpobfkfo' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource.

It still works in non-Incognito mode though.

After the Chrome update, it did not work anymore, but after updating TM to 4.12, it *did* work again... until I closed Chrome and re-opened it. Now it never works again, even after restarting Chrome or reloading the extension.

## Specifications

- Chromium: Version 90.0.4430.72 (Official Build) (64-bit)
- TM: 4.12
- OS: Windows 10

## Script

(Please give an example of the script if applicable.)

```js
// ==UserScript==
// @name         New Userscript
// @namespace    http://tampermonkey.net/
// @version      0.1
// @description  try to take over the world!
// @author       You
// @match        *://example.net/*
// @grant        GM_xmlhttpRequest
// ==/UserScript==

(function() {
    'use strict';

    GM_xmlhttpRequest({
      method: 'GET',
      url: window.location.href,
      onload: function(resp) {
        alert("OK (incognito=" + GM_info.isIncognito + ")");
      },
      onerror: function() {
        alert("ERROR (incognito=" + GM_info.isIncognito + ")");
      },
    });
})();
```

Install script then go to https://example.net/.

**Comment by jdoe1024** [comment]
I've just tried installing a fresh Chrome+TM in a Windows Sandbox, and the behavior is the same: `GM_xmlhttpRequest()` works in Incognito after TM install (and allowing TM in Incognito pages), but after relaunching Chrome, it does not work anymore.

I've also changed the example script target from example.org to example.net since the former is blacklisted.

**Comment by derjanb** [comment]
This is a Chrome 90+ issue. Unfortunately there is nothing I can do to fix or workaround it. 

https://bugs.chromium.org/p/chromium/issues/detail?id=1197798

**Comment by jdoe1024** [comment]
OK thanks, as it does work just after installing the extension, I thought it could be a bug in TM instead. Glad to see the Chrome dev team is working on it.

**Comment by jdoe1024** [comment]
Fixed in Chrome 90.0.4430.93 and confirmed.
